### ISSUE TITLE
Add the collapsible Visibility section

### ISSUE THREAD

**Issue by mdperez86** [woothemes/woocommerce#43705]
**Description**

Use the sidebar/modal to show some product settings from the Organization tab.
Note that gutenberg also has something similar for posts and pages already, see the [`PostVisibility`](https://github.com/WordPress/gutenberg/blob/0a5bbb4a07bab5b0f2ee0a1276339eda5d1a5a95/packages/editor/src/components/post-visibility/index.js#L20) component, we should try to re-use / mimic as much as we can from this component.

**Epic**

#40371

Depends on #43704

**Design (temporary link)**

[AkVNGImLgSqCObTQ3idVn7-fi-3359_344908](https://href.li/?https://www.figma.com/file/AkVNGImLgSqCObTQ3idVn7/?node-id=3359-344908)

**ACs**

```[tasklist]
### Tasks
- [ ] Create the collapsible Visibility section AkVNGImLgSqCObTQ3idVn7-fi-3359_345477
- [ ] Add `Hide in product catalog` checkbox like in the Organization tab
- [ ] Add `Hide from search results` checkbox like in the Organization tab
- [ ] Add `Enable product reviews` checkbox like in the Organization tab
- [ ] Add `Require a password` checkbox like in the Organization tab
- [ ] When the `Require a password` box is checked, the user can enter the password directly in the panel
- [ ] The section and the block in the Organization tab are fully synced
- [ ] The title `Visibility: Public` changes to `Visibility: Hidden` if `Hide in product catalog` or `Hide from search results` as checked (there’s also a badge in the title bar)
```

**Comment by mdperez86** [comment]
@j111q @pmcpinto 

>(there’s also a badge in the title bar)

I don't see it. Do we have the design for this?

**Comment by louwie17** [comment]
> @j111q @pmcpinto
> 
> > (there’s also a badge in the title bar)
> 
> I don't see it. Do we have the design for this?

I assume this is referring to this `Hidden` badge in the header:
<img width="671" alt="Screenshot 2024-01-16 at 4 15 21 PM" src="https://github.com/woocommerce/woocommerce/assets/2240960/c4564c88-2e11-4ab3-9d8e-bb63d54e5cad">

Figma link: AkVNGImLgSqCObTQ3idVn7-fi-3359_340432 

Although maybe the header updates are part of another issue?

**Comment by octaedro** [comment]
Hey @louwie17,
I couldn't find any issue with the header updates. I'll remove the badge from the scope of this issue since I think it should be tackled separately. If it's ok with you, I will create an issue to work on the header changes related to visibility.
